Automattic/wp-calypso
# Jetpack Connect: create a core Jetpack remote install flow and handle on success requests.

Flow starts at `/jetpack/connect` and redirects to `jetpack/connect/install` where the credentials form lives for .org sites with Jetpack not installed or activated. Uses a dedicated wpcom API endpoint
and selectors defined in https://github.com/Automattic/wp-calypso/pull/22555.

*Changes included:*
* adds feature flag, route definition for `jetpack/connect/install`,
* base UI for the credentials ( additional styling in the follow-up),
* handles **on success** responses ( valid credentials and admin permissions for the site),

*Not covered:*
- full styling for entire flow ( Worg logo in the credentials form, footer links from the mockups, new button branding, wizard counter ),

**To test:**
- Apply D10065 on the .com sandbox and sandbox.
- Checkout this branch on local Calypso and start at `calypso.localhost:3000/jetpack/connect/`
Input url of a remote site with:
1. Jetpack installed and activated- the flow should redirect to the JPC authorization form,
2. no Jetpack installed (or installed but not activated)- verify the redirect to the screen with form credentials at `jetpack/connect/install`,

- input valid site credentials,
- verify that Jetpack was installed (or activated if already installed) and the flow redirects to the JPC authorization step ( .com login and/or JP plans page).

![screen shot 2018-02-26 at 22 13 05](https://user-images.githubusercontent.com/13561163/36702060-81f69928-1b4d-11e8-98c6-a793e8319ae7.png)
